apache2/re_operators.c

```diff
@@ -1812,7 +1812,7 @@ static int msre_op_gsbLookup_execute(modsec_rec *msr, msre_rule *rule, msre_var
 
                         }
 
-                        url = apr_palloc(rule->ruleset->mp, strlen(canon));
+                        url = apr_pstrdup(rule->ruleset->mp, strlen(canon));
```

Comment:
We're passing `strlen(canon)` as `const char*`.  This should probably be the source string. [source](https://apr.apache.org/docs/apr/trunk/group__apr__strings.html#gabc79e99ff19abbd7cfd18308c5f85d47)

Reply:
apr_pstrdup seems to be the recommended method. We usages in the same file. Are you asking to pass the string as pointer?

Reply:
Check the link I've shared in my previous comment and compare the argument types with what you've passed.